sharktank/tests/models/t5/t5_test.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from transformers.models.t5.modeling_t5 import (
+    T5Attention as ReferenceT5Attention,
+    T5LayerSelfAttention as ReferenceT5LayerSelfAttention,
+    T5LayerFF as ReferenceT5LayerFF,
+)
+from transformers import (
+    AutoTokenizer,
+    T5EncoderModel as ReferenceT5EncoderModel,
+    T5Config as ReferenceT5Config,
+)
+import pytest
+import torch
+from unittest import TestCase
+from parameterized import parameterized
+from sharktank.types import Theta, DefaultPrimitiveTensor, unbox_tensor, Dataset
+from sharktank.models.t5.t5 import (
+    T5Attention,
+    T5SelfAttention,
+    T5Config,
+    T5Encoder,
+    T5LayerFF,
+)
+from sharktank.utils.testing import make_rand_torch, longrun
+from sharktank.utils.hf_datasets import get_dataset
+
+
+def make_random_mask(shape: tuple[int], dtype: torch.dtype):
+    mask = make_rand_torch(shape=shape, dtype=dtype)
+    mask = (mask >= 0).to(dtype=dtype)
+    return mask
+
+
+@pytest.mark.usefixtures("get_model_artifacts")
+class T5EncoderTest(TestCase):
+    def setUp(self):
+        super().setUp()
+        torch.random.manual_seed(12345)
+        torch.no_grad()
+
+    @parameterized.expand(
+        [
+            "google/t5-v1_1-small",
+            "google/t5-v1_1-xxl",
+        ]
+    )
+    @longrun
+    def testV1_1Fp32CompareTorchEagerAgainstHuggingFace(self, huggingface_repo_id: str):
+        get_dataset(
+            huggingface_repo_id,
+        ).download()
```

Comment:
should we just cache the model for the CI instead of downloading each time? May end up with failures due to corrupted downloads on occasion.